provider/pkg/provider/deployment_settings.go

```diff
@@ -52,10 +54,47 @@ func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap() resource.Propert
 				if ds.SourceContext.Git.GitAuth.SSHAuth != nil {
 					sshAuthPropertyMap := resource.PropertyMap{}
 					if ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey.Value != "" {
-						sshAuthPropertyMap["sshPrivateKey"] = resource.NewPropertyValue(FixMe)
+						if plaintextSettings == nil {
+							importSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+						} else {
+							if oldCipherSettings == nil {
+								createSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey,
+									plaintextSettings.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
+							} else {
+								if oldCipherSettings.SourceContext != nil &&
+									oldCipherSettings.SourceContext.Git != nil &&
+									oldCipherSettings.SourceContext.Git.GitAuth != nil &&
+									oldCipherSettings.SourceContext.Git.GitAuth.SSHAuth != nil {
```

Comment:
Really wish we didn't have to do this :(

Reply:
Fully agree, I hated every second of typing this out, but no way around in go that I know of...
I really miss `?.`